### ISSUE TITLE
how to do inline fragments?

### ISSUE THREAD

**Issue by capaj** [acro5piano/typed-graphqlify#9]
on our Graphql api we use interfaces. To query an interface we need to make inline fragments such as:
```graphql
query HeroForEpisode($ep: Episode!) {
  hero(episode: $ep) {
    name
    ... on Droid {
      primaryFunction
    }
    ... on Human {
      height
    }
  }
}
```

if this lib can support these we can start putting it to production at https://www.looop.co/

**Comment by acro5piano** [comment]
Production application is the goal to OSS!
I haven't used inline fragments, so take a look tomorrow.

Thanks,

**Comment by capaj** [comment]
huh now that I am thinking about this-we could use the same approach I was suggesting for #2 and do:

```ts
import {fragment} from 'typed-graphqlify'
const queryObject = {
  hero: fragment({name: types.string}, {
    Droid: {
      primaryFunction: types.string,
    },
    Human: {
      height: types.number,
    }
  })
}
``` 

so fragment would take 2 params-first would be common fields and the second would have interface types and their respective fields. It should be possible to express the return type of that in typescript I think.

**Comment by acro5piano** [comment]
I looked at your changes carefully, and found it could divided to the following features:

Already solved:

- Fixing CJS format https://github.com/acro5piano/typed-graphqlify/pull/13
- Make operation name optional https://github.com/acro5piano/typed-graphqlify/pull/15

Not solved:

- Fixing README
- Refactor code by adding `CoreType` and `TypeFlags`
- Enables to infer operation variables by Adding
    - `params` and `$` operation helpers
    - `types.raw` expression
    - `paramsWeakMap` logic (cool!)

**For `Fixing README`**

If you create a PR, I will merge.
I don't do it on my own because I would like to keep that you contributed the code and cannot cherry-pick the commit.

**For `Refactor code adding CoreType and TypeFlags`**

After I look at the code, and will merge. I already saw the code, so maybe I will merge it soon.

**For `Enables to infer operation variables`**

I think there should be discussion to merge this feature, because I would like to keep this library simple as possible.
Query in TypeScript should be similar to raw GraphQL, in order to enable developers to migrate easily,
So the possible solution is that make this feature is optional, while providing simpler way to write parameters without using `__params` field.

However, it is really cool feature to inter operation variables from query variables, so it is really awesome if you (or anyone) implement this feature.

So I will close this PR. Sorry.
But your `Fixing CJS format` commit is kept as your contribution, as I cherry-picked your commit :)

I will work on this issue https://github.com/acro5piano/typed-graphqlify/issues/9 in a while.
